src/bin/winkernel/driver.c

```diff
@@ -143,6 +145,7 @@ Return Value:
         MsQuicLibraryUnload();
     }
 
+    QuicSiloDetatch(PrevSilo);
```

Comment:
Does `DriverUnload` get called when `DriverEntry` returns error?

Reply:
Nope. Is that an issue?

Reply:
If you register an ETW provider and then unload the image (dll/sys) that contains the enable callback without unregistering the provider, there will be a crash when the callback is next invoked.

Reply:
(I don't know if that's happening here or not, but just FYI)

Reply:
We have proper error handling in DriverEntry to ensure everything gets cleaned up when bailing out.